src/components/userForms/UserRegistrationForm.tsx

```diff
@@ -0,0 +1,151 @@
+import { SubmitHandler, useForm } from "react-hook-form";
+import { UserRegistrationFormValues } from "./UserRegistrationForm.types.ts";
+import { StyledForm } from "../forms/Form.styles.ts";
+import LabelField from "../formElements/LabelField.tsx";
+import InputField from "../formElements/InputField.tsx";
+import { ErrorMessage } from "@hookform/error-message";
+import Text from "../text/Text.tsx";
+import FormButton from "../forms/FormButton.tsx";
+import { pb, USERS_COLLECTION } from "../../Backend.constants.ts";
+import { toast } from "react-toastify";
+import { useNavigate } from "react-router-dom";
+
+const UserRegistrationForm = () => {
+  const navigateTo = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    watch,
+    setError,
+    formState: { errors },
+  } = useForm<UserRegistrationFormValues>({ mode: "onChange" });
+
+  const submitUserData: SubmitHandler<UserRegistrationFormValues> = async (
```

Comment:
I'm happy you can handle side effects both ways. 
If you have patience for it, I'd rewrite it using react-query ;)

Reply:
is this a better approach? using react query?